rocketpy/simulation/flight.py

```diff
@@ -1185,6 +1213,19 @@ def __init_solver_monitors(self):
         self.t = self.solution[-1][0]
         self.y_sol = self.solution[-1][1:]
 
+        if isinstance(self.ode_solver, OdeSolver):
+            self._solver = self.ode_solver
+        else:
+            try:
+                self._solver = ODE_SOLVER_MAP[self.ode_solver]
+            except KeyError as e:
+                raise ValueError(
+                    f"Invalid ``ode_solver`` input: {self.ode_solver}. "
+                    f"Available options are: {', '.join(ODE_SOLVER_MAP.keys())}"
+                ) from e
+
+        self.__is_lsoda = hasattr(self._solver, "_lsoda_solver")
+
```

Comment:
Quite verbose...

Maybe we should create a separate method for this validation.

Reply:
Fixed by [26ceab9](https://github.com/RocketPy-Team/RocketPy/pull/748/commits/26ceab9dd04f5676dd357e134e9dc88bd96c4c5c).